packages/builder/src/package.ts

```diff
@@ -76,12 +76,23 @@ export class PackageReference {
 
     if (!match || !match.groups?.name) {
       throw new Error(
-        `Invalid package name "${ref}". Should be of the format <package-name>:<version> or <package-name>:<version>@<preset>`
+        `Invalid package reference "${ref}". Should be of the format <package-name>:<version> or <package-name>:<version>@<preset>`
       );
     }
 
     const res: PartialRefValues = { name: match.groups.name };
 
+    const nameSize = new Blob([res.name]).size;
+    const variantSize = new Blob([res.version + '_' + res.preset]).size;
```

Comment:
no need to use `Blob.size` here, as the regex makes sure that there are no values that occupy more than 1 char because the only possible chars are `[a-z0-9-_]`, so `res.name.length` should work correctly AFAIK